nixos/lib/test-driver/test-driver.py

```diff
@@ -675,6 +676,26 @@ def screen_matches(last: bool) -> bool:
         with self.nested("waiting for {} to appear on screen".format(regex)):
             retry(screen_matches)
 
+    def wait_for_console_text(self, regex: str) -> None:
+        self.log("waiting for {} to appear on console".format(regex))
+        screen = []  # store last 100 lines, ~ a screen
+        while True:
+            try:
+                line = self.last_lines.get()
+            except queue.Empty:
+                self.sleep(1)
+                continue
+            matches = re.search(regex, line)
+            if matches is not None:
+                return
+            screen.append(line)
+            if len(screen) == 100:
+                text = "\n> ".join(screen)
+                screen = []
+                self.log(
+                    "Last attempt failed. Text was:\n===\n> {}\n===\n".format(text)
```

Comment:
assuming i have 1000 lines in the buffer, this would fill 10 screens. if i excpect my keyword to reside very late in one of the last 1000 lines, then i will get 9 log lines that say "last attempt failed", although i would not consider this an error. Or am i completely off here?

Reply:
That's correct. Do you think it's too verbose? I copied the approach of `wait_for_text` but it may not really appropriate in this case.

Reply:
Dropped.